docker-compose.yml

```diff
@@ -1,38 +1,81 @@
 version: "3"
 
 services:
+  redis:
+    image: redis:5
+    restart: unless-stopped
+    ports:
+      - "6379:6379"
+    healthcheck:
+      test: ['CMD', 'redis-cli', 'ping']
+      interval: 1s
+      timeout: 3s
+      retries: 30
+
+  elasticsearch:
+    image: kuzzleio/elasticsearch:7
+    restart: unless-stopped
+    ports:
+      - "9200:9200"
+    environment:
+      - ingest.geoip.downloader.enabled=false
+    healthcheck:
+      test: ['CMD', 'curl', '-f', 'http://localhost:9200']
+      interval: 2s
+      timeout: 2s
+      retries: 10
+    ulimits:
+      nofile: 65536
+    volumes:
+      - esdata01:/usr/share/elasticsearch/data
+
   kuzzle:
-    image: kuzzleio/kuzzle-runner:14
-    command: ${DOCKER_COMMAND:-npm run dev}
+    image: kuzzleio/kuzzle-runner:16.18.0
+    restart: unless-stopped
+    command: "bash /docker/start-backend.sh"
+    profiles: ["backend"]
+    depends_on:
+      redis:
+        condition: service_healthy
+      elasticsearch:
+        condition: service_healthy
     cap_add:
       - SYS_PTRACE
     ulimits:
       nofile: 65536
     sysctls:
       - net.core.somaxconn=8192
-    volumes:
-      - .:/var/app
     ports:
       - "7512:7512"
       - "9229:9229"
       - "1883:1883"
-    depends_on:
-      - redis
+    healthcheck:
+      test: ['CMD', 'curl', '-f', 'http://kuzzle:7512/_healthCheck']
```

Comment:
Since you are using Kuzzle 2.25, you can use "http://localhost:7512/_healthcheck" new public route

Reply:
I think there is mistake because, I have used, this route for healthcheck on kuzzle service.

Reply:
http://kuzzle:7512/_healthCheck does exists indeed but it will be deprecated

Reply:
Hum sorry, I Don't really understand the difference between "http://localhost:7512/_healthcheck" and "http://kuzzle:7512/_healthCheck" for the test.
The only difference is the DNS and I found it's more accurate to use the service name instead localhost which is less clear.

Could you explain to me the difference ?

Reply:
Sorry, I tested locally then forgot to update with the service name indeed

I meant http://kuzzle:7512/_healthcheck

The route you are using http://kuzzle:7512/_healthCheck is not a public route, it will be working as is until your kuzzle instance allow the anonymous the use this controller server.

```
{
--
verb: "get",
path: "/_healthCheck",
controller: "server",
action: "healthCheck",
}
```

Whereas the http://kuzzle:7512/_healthcheck and will always return a 200 since it is a public route

Reply:
Ok my bad, I just saw the difference in typo between `_healthCheck` and `_healthcheck`

Reply:
changed